src/extension/messages/platformRequest/saveCellExecution.ts

```diff
@@ -25,10 +26,18 @@ export default async function saveCellExecution(
   kernel: Kernel,
 ): Promise<void | boolean> {
   const { messaging, message, editor } = requestMessage
+  // Save the file to ensure the serialization completes before saving the cell execution.
+  // This guarantees we access the latest cache state of the serializer.
+  await editor.notebook.save()
+
   log.info('Saving cell execution')
 
   const escalationButton = kernel.hasExperimentEnabled('escalationButton', false)!
   const sessionId = kernel.getRunnerEnvironment()?.getSessionId()
+  const cacheId = GrpcSerializer.getDocumentCacheId(editor.notebook.metadata) as string
+  const plainSessionOutput = await kernel.getSerializer()?.getPlainCache(cacheId)
```

Comment:
Why not make `getPlainCache` and `getMaskedCache` public methods on `Kernel`? I don't love exposing the `serializer` instance directly.

Reply:
Fixed in 2c69b84a17dcf327bd3ee5e4e305200fb220cece